### ISSUE TITLE
Presence re-entry requirement change for 1.1

### ISSUE THREAD

**Issue by paddybyers** [ably/ably-cocoa#878]
See https://github.com/ably/docs/pull/711.

**Comment by ricardopereira** [comment]
@paddybyers I'm reviewing this and I think it's updated but maybe I'm missing something. So, on every SYNC message (`RTP17c1`) the library will do a [sync](https://github.com/ably/ably-cocoa/blob/f3017852ec5a9a34ef3da72cf7b8fe5429903d89/Source/ARTRealtimeChannel.m#L873) and if the message is the last serial then an [automatic re-entry is performed](https://github.com/ably/ably-cocoa/blob/f3017852ec5a9a34ef3da72cf7b8fe5429903d89/Source/ARTPresenceMap.m#L179-L184).

Now, the library does a sync when the ATTACHED message has false in `HAS_PRESENCE` flag (`RTP17c2`) but it checks some extra checks:

https://github.com/ably/ably-cocoa/blob/f3017852ec5a9a34ef3da72cf7b8fe5429903d89/Source/ARTRealtimeChannel.m#L698-L708

Is this still valid?
Or the library should do an automatic re-entry every time the `HAS_PRESENCE` is false?

**Comment by ricardopereira** [comment]
@paddybyers bump ^

**Comment by AndyNicks** [comment]
@paddybyers Hey Paddy, can you have a look at Ricardo's comment please

**Comment by lukasz-szyszkowski** [comment]
As far as I understand it we shouldn't do any additional checks like here:
https://github.com/ably/ably-cocoa/blob/48cb1c6f103fb5a7192c1463b1c327ecfafe3642/Source/ARTRealtimeChannel.m#L735-L737
according to [RTP17f](https://sdk.ably.com/builds/ably/specification/main/features/#RTP17f) 
```
The RealtimePresence object should perform automatic re-entry whenever a channel moves into the ATTACHED state. (It does not need to do so for RTL12 ATTACHED events received on already-ATTACHED channels).
```
Am I correct @paddybyers?

**Comment by lawrence-forooghian** [comment]
@maratal is currently working on #1494, which updates the library to use protocol version 2 and specification version 2.0.0. This specification version simplifies the criteria for performing an automatic re-entry, replacing RTP17c with [RTP17f](https://sdk.ably.com/builds/ably/specification/main/features/#RTP17f). I would suggest that we park the current bug and close it once we've implemented RTP17f as part of #1494.

**Comment by lawrence-forooghian** [comment]
Hey @maratal, would you mind confirming whether you agree with my above assessment, that #1494 will resolve this issue? I'll assign this one to you, to close after #1494. If you disagree with my assessment feel free to assign back to me.

**Comment by maratal** [comment]
@lawrence-forooghian Correct, RTP17c was replaced with RTP17f and implemented in https://github.com/ably/ably-cocoa/issues/1494
